streams/src/main/java/org/apache/kafka/streams/processor/TimestampExtractor.java

```diff
@@ -27,17 +27,19 @@
 public interface TimestampExtractor {
 
     /**
-     * Extracts a timestamp from a record.
+     * Extracts a timestamp from a record. The timestamp must be positive. If a negative timestamp is returned,
```

Comment:
Agreed. I would like to write `The timestamp must be non-negative` which includes zero, but this is hard to parse to the user... But `The timestamp must be zero or positive` does also sound clumsy... Not sure how to phrase is nicely... Any suggestion?